tests/benchmarks/cpu/intersect_all.cpp

```diff
@@ -0,0 +1,89 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2020-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s)
+#include "detray/core/detector.hpp"
+#include "detray/detectors/create_toy_geometry.hpp"
+#include "detray/detectors/detector_metadata.hpp"
+#include "detray/intersection/intersection_kernel.hpp"
+#include "detray/simulation/event_generator/track_generators.hpp"
+#include "detray/tracks/tracks.hpp"
+#include "detray/utils/ranges.hpp"
+
+// Vecmem include(s)
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// Google Benchmark include(s)
+#include <benchmark/benchmark.h>
+
+// System include(s)
+#include <iostream>
+#include <map>
+#include <string>
+
+// This test runs intersection with all surfaces of the TrackML detector
+void BM_INTERSECT_ALL(benchmark::State &state) {
+
+    static const unsigned int theta_steps{100u};
+    static const unsigned int phi_steps{100u};
+
+    // Detector configuration
+    static constexpr std::size_t n_brl_layers{4u};
+    static constexpr std::size_t n_edc_layers{7u};
+    vecmem::host_memory_resource host_mr;
+    auto d = detray::create_toy_geometry(host_mr, n_brl_layers, n_edc_layers);
+
+    decltype(d)::geometry_context geo_context;
+
+    const auto &masks = d.mask_store();
+    const auto &transforms = d.transform_store(geo_context);
+
+    unsigned int hits{0u};
+    unsigned int missed{0u};
+
+    for (auto _ : state) {
+        __plugin::point3<detray::scalar> pos{0.f, 0.f, 0.f};
+
+        // Iterate through uniformly distributed momentum directions
+        for (const auto track :
+             detray::uniform_track_generator<detray::free_track_parameters<
+                 __plugin::transform3<detray::scalar>>>(theta_steps, phi_steps,
+                                                        pos)) {
+
+            // Loop over volumes
+            for (const auto &v : d.volumes()) {
+                // Loop over all surfaces in volume
+                for (const auto &sf : d.surfaces(v)) {
+
+                    auto sfi =
+                        masks.template visit<detray::intersection_update>(
+                            sf.mask(), detray::detail::ray(track), sf,
+                            transforms);
+
+                    benchmark::DoNotOptimize(hits);
+                    benchmark::DoNotOptimize(missed);
+                    if (sfi.status == detray::intersection::status::e_inside) {
+                        ++hits;
+                    } else {
+                        ++missed;
+                    }
+                }
+            }
+        }
+    }
+
+#ifdef DETRAY_BENCHMARK_PRINTOUTS
```

Comment:
I ended up introducing a separate cache variable / macro for printing these "status messages". Making all of the benchmarks behave the same in the process.

Reply:
So, we can have these outputs on a multithreaded benchmark? Won't that just spam the output?